docs/user-guide/ui.md

```diff
@@ -56,51 +57,111 @@ Normally, this should be taken care of automatically by the
 
-If needed, you can expand the "more options" section where you can change more advanced settings,
-such as the Kubernetes namespace that the resulting Pods run in, image pull secrets for private
-registries, resource limits, container entrypoint and privileged status.
+#### Specifying application details
+
+The wizard expects that you provide the following information:
+
+- **App name** (mandatory): Name for your application. A [label](http://kubernetes.io/v1.1/docs/user-guide/labels.html) with the name will be added to the Replication Controller and Service, if any, that will be deployed.
+
+  The application name must be unique within the selected Kubernetes namespace. It must start with a lowercase character, and contain only lowercase letters, numbers and dashes (-). It is limited to 24 characters.
+
+- **Container image** (mandatory): The URL of a public Docker [container image](http://kubernetes.io/v1.1/docs/user-guide/images.html) on any registry, or a private image (commonly hosted on the Google Container Registry or Docker Hub).
+
+- **Number of pods** (mandatory): The target number of Pods you want your application to be deployed in. The value must be a positive integer.
+
+  A [Replication Controller](http://kubernetes.io/v1.1/docs/user-guide/replication-controller.html) will be created to maintain the desired number of Pods across your cluster.
+
+- **Ports** (optional): If your container listens on a port, you can provide a port and target port. The wizard will create a corresponding Kubernetes [Service](http://kubernetes.io/v1.1/docs/user-guide/services.html) which will route to your deployed Pods. Supported protocols are TCP and UDP. In case you specify ports, the internal DNS name for this Service will be the value you specified as application name above.
+
+  Be aware that if you specify ports, you need to provide both port and target port.
+
+- For some parts of your application (e.g. frontends), you can expose the Service onto an external, maybe public IP address by selecting the **Expose service externally** option. You may need to open up one or more ports to do so. Find more details [here](http://kubernetes.io/v1.1/docs/user-guide/services-firewalls.html).
+
+If needed, you can expand the **Advanced options** section where you can specify more settings:
 
 ![Kubernetes Dashboard deploy form advanced options](ui-dashboard-deploy-more.png)
 
-#### Applications view
+- **Description**: The text you enter here will be added as an [annotation](http://kubernetes.io/v1.1/docs/user-guide/annotations.html) to the Replication Controller and displayed in the application's details.
+
+- **Labels**: Default [labels](http://kubernetes.io/v1.1/docs/user-guide/labels.html) to be used for your application are application name and version. You can specify additional labels to be applied to the Replication Controller, Service (if any), and Pods, such as release, environment, tier, partition, and release track.
+
+  Example:
+``
```

Comment:
Wrap the example in three upticks instead of two. This will fix leading spaces and made the example a block content, instead of inline one.